sql/core/src/main/scala/org/apache/spark/sql/execution/joins/ShuffledHashJoinExec.scala

```diff
@@ -83,8 +88,10 @@ case class ShuffledHashJoinExec(
       iter,
       buildBoundKeys,
       taskMemoryManager = context.taskMemoryManager(),
-      // Full outer join needs support for NULL key in HashedRelation.
-      allowsNullKey = joinType == FullOuter,
+      // build-side outer join needs support for NULL key in HashedRelation.
```

Comment:
Isn't the original comment still valid?

Reply:
Hi @viirya , i changed from :

```// Full outer join needs support for NULL key in HashedRelation.```
to
```// build-side outer join needs support for NULL key in HashedRelation.```

as now its a more general use-case that need the null key in HashedRelation, not just full outer join

Reply:
`build-side outer join` sounds like left-outer or right-outer join specially. For full outer join, right/left side both can be build side, no?

Reply:
I see how it may be read that way, let me clarify it.  I was originally hoping to find one phrase to capture full-outer join, right-outer + buildRight , left-outer + buildLeft

Reply:
Maybe "build side or full outer join"

Reply:
OK i changed all the methods, and comments , hope it is more clear.

Reply:
Thank you.